Hello/Block/Main.php

```diff
@@ -1,24 +1,55 @@
 <?php
 namespace Excellence\Hello\Block;
+ use Magento\Framework\Controller\ResultFactory;
   
 class Main extends \Magento\Framework\View\Element\Template
 {   
+    protected $registry;
     protected $_testFactory;
     public function __construct(
+       \Magento\Framework\Message\ManagerInterface $messageManager,
         \Magento\Framework\View\Element\Template\Context $context,
-        \Excellence\Hello\Model\TestFactory $testFactory
+        \Excellence\Hello\Model\TestFactory $testFactory,
+        \Magento\Framework\Registry $registry
     )
     { 
         $this->testFactory=$testFactory;
+        $this->messageManager=$messageManager;
+         $this->registry = $registry;
         parent::__construct($context);
     }
     protected function _prepareLayout()
     {
-          $data=$this->testFactory->create();
-          $collectionData = $data->getCollection()->getData();
-         
-    $this->setCollectModel($collectionData);
+        $data=$this->testFactory->create();
+          $search = $this->registry->registry('search');
+          if(!empty($search)){
+           
+            $collectionData = $data->getCollection()
```

Comment:
dont write sql related things in block files. all sql related code should to go in model